.github/workflows/test.yml

```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: [ '8', '11' ]
+        java: [ '17' ]
```

Comment:
Q: Why do we only test Java 17? Is there a special reason for that?

Reply:
Using `languageVersion = JavaLanguageVersion.of(8)`, I think we don't need to cate the gradle execution JDK version too much, so just use 17 is enough IIUC.

Reply:
We can use Java 8 for compilation but I think it is necessary to run tests on Java 8, 11, 17 and the next LTS. JDK can deprecate a method with `forRemoval=true` which we need to detect in CI builds.

Reply:
Understood, I added `testOn11` and `testOn17` tasks, so it will use specific toolchain version to run tests.